src/test/java/io/vertx/ext/couchdb/CouchdbClientTest.java

```diff
@@ -1,31 +1,58 @@
 package io.vertx.ext.couchdb;
 
-import static org.junit.jupiter.api.Assertions.*;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
 import io.vertx.core.Future;
 import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
 import io.vertx.core.json.JsonArray;
 import io.vertx.core.json.JsonObject;
-import io.vertx.ext.auth.authentication.UsernamePasswordCredentials;
+import io.vertx.ext.couchdb.impl.CouchdbClientImpl;
 import io.vertx.ext.couchdb.utils.UnitTest;
-import io.vertx.ext.web.client.WebClientOptions;
+import io.vertx.ext.web.client.HttpRequest;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.junit5.VertxTestContext;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+
+import java.lang.reflect.Field;
+import java.util.concurrent.TimeUnit;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.BAD_REQUEST;
+import static io.netty.handler.codec.http.HttpResponseStatus.CREATED;
+import static io.netty.handler.codec.http.HttpResponseStatus.PRECONDITION_FAILED;
+import static io.netty.handler.codec.http.HttpResponseStatus.UNAUTHORIZED;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
 
 @UnitTest
 class CouchdbClientTest {
 
-  Vertx vertx;
-  WebClientOptions options;
-  CouchdbClient client = null;
+  @Mock
+  private WebClient mockWebClient;
+
+  @Mock
+  private HttpRequest<Buffer> mockHttpRequest;
+
+  @Mock
+  private HttpResponse<Buffer> mockHttpResponse;
+
+  private CouchdbClient client;
 
   @BeforeEach
-  void beforeEach() {
-    this.vertx = Vertx.vertx();
-    this.options = new WebClientOptions();
-    this.options.setDefaultHost("localhost")
-        .setDefaultPort(5984);
-    client = CouchdbClient.create(vertx, options);
-    client.authetication(new UsernamePasswordCredentials("admin", "password"));
+  void setUp() throws Exception {
+    Vertx vertx = Vertx.vertx();
```

Comment:
Use `@ExtendWith(VertxExtension.class)` -> add to the `@IntregrationTest` annotation

Reply:
Already added this at `@IntegrationTest`annotation. Would just need to apply it.